pkg/controller/ecs/service/setup.go

```diff
@@ -100,10 +100,8 @@ func preCreate(_ context.Context, cr *svcapitypes.Service, obj *svcsdk.CreateSer
 
 func preDelete(_ context.Context, cr *svcapitypes.Service, obj *svcsdk.DeleteServiceInput) (bool, error) {
 	if aws.StringValue(cr.Spec.ForProvider.SchedulingStrategy) == svcsdk.SchedulingStrategyReplica {
-		if aws.Int64Value(cr.Status.AtProvider.RunningCount) > 0 || aws.Int64Value(cr.Status.AtProvider.PendingCount) > 0 {
-			// Force deletion of services even if they are not scaled down.
-			obj.SetForce(true)
```

Comment:
What are the implications if the user does NOT want to enforce deletion on running instances? I wonder if it wouldn't be better if there's an explicit `spec.forProvider.forceDeletion` property?

Reply:
Your proposal is to add the field `ForceDeletion` at `apis/ecs/v1alpha1/zz_service.go` struct `ServiceParameters` and in the `preDelete` func do the obj.SetForce with the value in CR?

Reply:
Yes, I think that would make things more explicit and avoid a potential breaking behaviour for existing resources.

Reply:
Could you please give guidance on how to add the field, I have tried to change the `generator-config.yaml`, but without success.

Reply:
APIs that are generated with ACK have a `custom_types.go` that allow you to add additional properties to `spec.forProvider`. For `ecs` you can find a [`CustomServiceParameters`](https://github.com/crossplane-contrib/provider-aws/blob/master/apis/ecs/v1alpha1/custom_types.go#L89) struct that you can extend.

You then have to regenerate the methods and CRDs in `package` using `make generate`.

You can find more info here: https://github.com/crossplane-contrib/provider-aws/blob/master/CODE_GENERATION.md